fe/src/main/java/org/apache/doris/load/loadv2/dpp/SparkRDDAggregator.java

```diff
@@ -0,0 +1,194 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.load.loadv2.dpp;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.doris.load.loadv2.BitmapValue;
+import org.apache.doris.load.loadv2.etl.EtlJobConfig;
+import org.apache.spark.api.java.function.Function2;
+import org.apache.spark.api.java.function.PairFunction;
+import org.apache.spark.sql.Row;
+import scala.Tuple2;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+
+public abstract class SparkRDDAggregator<T> implements Serializable {
+
+    abstract T aggregate(T v1, T v2);
+
+    abstract byte[] serialize(Object value);
+
+    T buildValue(Object value) {
+        return (T) value;
+    }
+
+    public static SparkRDDAggregator buildAggregator(EtlJobConfig.EtlColumn column) {
```

Comment:
throws exception

Reply:
Do you mean we need a UserException instead of a RuntimeException?